libdnf/utils/utils_internal.hpp

```diff
@@ -38,6 +43,20 @@ static inline bool is_file_pattern(const std::string & pattern) {
     return pattern[0] == '/' || (pattern[0] == '*' && pattern[1] == '/');
 }
 
+inline Id id_to_lowercase_id(Pool * pool, const char * name_cstring, int create) {
+    int name_length = static_cast<int>(strlen(name_cstring));
+    auto tmp_name_cstring = pool_alloctmpspace(pool, name_length);
+    for (int index = 0; index < name_length; ++index) {
+        tmp_name_cstring[index] = static_cast<char>(tolower(static_cast<unsigned char>(name_cstring[index])));
```

Comment:
Why are you cast `char` to `unsigned char` (`tolower(static_cast<unsigned char>(name_cstring[index]))`)?
`tolower` takes `int`. So, `tolower(static_cast<int>(name_cstring[index]))` is better.
Or simply `tolower(name_cstring[index])` without casting.